### ISSUE TITLE
Mixer crash looping OOM, prometheus stuck.

### ISSUE THREAD

**Issue by costinm** [istio/istio#7734]
On my long running cluster, noticed that telemetry server was crashing with 'out of memory' after jumping to 32G of RAM every 3 min. After investigation, found that prometheus logs had:

```
 level=warn ts=2018-08-08T14:02:39.484736721Z caller=scrape.go:717 component="scrape manager" scrape_pool=kubernetes-cadvisor target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-xrfd/proxy/metrics/cadvisor msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.509560231Z caller=scrape.go:713 component="scrape manager" scrape_pool=kubernetes-cadvisor target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-0t1n/proxy/metrics/cadvisor msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.510489755Z caller=scrape.go:717 component="scrape manager" scrape_pool=kubernetes-cadvisor target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-0t1n/proxy/metrics/cadvisor msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.610082066Z caller=scrape.go:713 component="scrape manager" scrape_pool=kubernetes-cadvisor target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-hbwk/proxy/metrics/cadvisor msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.611487884Z caller=scrape.go:717 component="scrape manager" scrape_pool=kubernetes-cadvisor target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-hbwk/proxy/metrics/cadvisor msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.851037004Z caller=scrape.go:713 component="scrape manager" scrape_pool=kubernetes-nodes target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-qc84/proxy/metrics msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory"
level=warn ts=2018-08-08T14:02:39.910879654Z caller=scrape.go:713 component="scrape manager" scrape_pool=kubernetes-nodes target=https://kubernetes.default.svc:443/api/v1/nodes/gke-weekly-10-default-pool-5675810c-8dxf/proxy/metrics msg="append failed" err="WAL log samples: log series: write data/wal/000460: cannot allocate memory" 

```
Not sure if prometheus is the cause - or there are 2 separate bugs. 

Mixer log shows normal traffic - no errors that I can see, except the out of memory.
Both have been running for 1 week - prometheus without restart (11days).

istio-testing project, weekly10 clustesr.

**Comment by costinm** [comment]
Restarting prometheus: metrics show up again. So far mixer is fine - suspect the problem was caused by a mixer overloading, causing a cascading failure ( the other 2 replicas got overloaded, and kept going ). 
Mixer seems to also behave badly under memory pressure - if the load is above a certain level, GC can't 
keep up and memory keeps growing until it gets killed ( same as what was happening in pilot ).

For mixer the solution is simpler - just need to determine the limit based on load tests, and start dropping
requests when the limit is reached ( can be made a config option ). And alerting user via metrics/alerts to 
increase the replica count.  And reduce allocations to increase the limit.

**Comment by douglas-reid** [comment]
I'd be cautious about suggesting that the solution is "simple". Mixer operates in multiple modes (check/quota v. report). Dropping requests may or may not be acceptable depending on the mode and backend requirements (billing systems v dashboards). Reducing the overhead of handling requests, to the extent that is possible, sounds like a good idea.

It is worth noting that the Prometheus server's performance and stability are separate from Mixer entirely. Whatever Prometheus server does or fails to do will not impact Mixer in any fashion.

**Comment by munrodg** [comment]
Closed duplicate of #9694
